R/R/install.R

```diff
@@ -0,0 +1,154 @@
+#' Install Metaflow Python package
+#'
+#' This function wraps installation functions from [reticulate][reticulate::reticulate] to install the Python packages
+#' **metaflow** and it's Python dependencies.
+#'
+#' This package uses the [reticulate][reticulate::reticulate] package
+#' to make an interface with the [Metaflow](https://metaflow.org/)
+#' Python package.
+#'
+#' @param method `character`, indicates to use `"conda"` or `"virtualenv"`.
+#' @param prompt boolean, whether or not to prompt user for confirmation before installation. Default is TRUE.
+#' @param version `character`, version of Metaflow to install. The default version
+#' is the latest available on PyPi.
+#' @param ... other arguments sent to [reticulate::conda_install()] or
+#'    [reticulate::virtualenv_install()]
+#'
+#' @seealso
+#' [reticulate: Using reticulate in an R Package](https://rstudio.github.io/reticulate/articles/package.html),
+#' [reticulate: Installing Python Packages](https://rstudio.github.io/reticulate/articles/python_packages.html)
+#' @examples
+#' \dontrun{
+#' # not run because it requires Python
+#' install_metaflow()
+#' }
+#' @export
+install_metaflow <- function(method = c("conda", "virtualenv"),
+                             prompt = TRUE,
+                             version = NULL,
+                             ...) {
+  envname <- pkg.env$envname
+
+  env_set <- check_environment(envname)
+  if (method == "conda" && env_set[["virtualenv"]]) {
+    message("A virtualenv environment is detected for Metaflow.")
+    message("Can't install within a virtualenv using conda.")
+    stop("Please remove existing virtualenv by `remove_metaflow()` and run the install again.")
```

Comment:
I am trying to understand the rationale here for recommending the user to remove the virtualenv. My assumption is that the user is already inside a virtualenv because they might need access to some of the packages in that virtualenv through reticulate. But now through our messaging, we are indicating that the user should remove that virtualenv before they can proceed to install metaflow-r (while just executing `install_metaflow(method='virtualenv')` will get them the desired end state). 

Also, `remove_metaflow()` signals that executing this command will somehow remove metaflow from my system, but it instead removes the environment that might contain much more than just Metaflow. It might be a good idea to change the name of the function to say, `remove_metaflow_environment()` or something similar. If we go down that path, then let's modify the messages in that function to read - `No conda installation found. Aborting...` and `Cancelled...` instead of the current messages.